src/crimson/osd/pg_recovery.cc

```diff
@@ -544,7 +534,13 @@ void PGRecovery::enqueue_push(
   }).then_interruptible([this, obj] {
     logger().debug("enqueue_push:{}", __func__);
     using BackfillState = crimson::osd::BackfillState;
-    start_backfill_recovery(BackfillState::ObjectPushed(std::move(obj)));
+    if (backfill_state->is_triggered()) {
+      backfill_state->post_event(
+       BackfillState::ObjectPushed(std::move(obj)).intrusive_from_this());
+    } else {
```

Comment:
Do we need the `else`? Perhaps `post_event` alone might be enough.

Reply:
Yes, we need the `else`. We need to process the event immediately if the state machine of `BackfillState` is not processing any event, otherwise the event may not be processed at all. I added the `else` after meeting with one such error.